contracts/Groth16.sol

```diff
@@ -6,9 +6,17 @@ struct G1Point {
   uint y;
 }
 
+// A field element F_{p^2} encoded as `real + i * imag`.
+// We chose to not represent this as an array of 2 numbers, because both Circom
+// and Ethereum EIP-197 encode to an array, but with conflicting encodings.
+struct Fp2Element {
+  uint real;
```

Comment:
Just a minor thing: Earlier in another PR we had a discussion about that using `uint256` for these numbers maybe better conveys the intent of these really being big numbers.